api/test/views/mock_keystone.py

```diff
@@ -0,0 +1,191 @@
+import json
+
+from rest_framework import status
+from rest_framework.response import Response
+from django.http import Http404
+from rest_framework.views import APIView
+from django.http import JsonResponse, HttpResponse
+
+from threepio import logger
+
+from atmosphere import settings
+from atmosphere.settings.local import AUTHENTICATION as auth_settings
+from atmosphere.settings.local import TEST as test_settings
+
+
+class MockKeystoneView(APIView):
+    """
+    This is just a simple view to catch any url we are not handling
+    """
+
+    def get(self, request):
+        logger.info(" MockKeystoneView:get called")
+        for header in request.META:
+            logger.info("   header: %s   value %s" % (header, request.META[header]))
+        return JsonResponse({'text': 'get response'})
+
+    def post(self, request):
+        logger.info("MockKeystoneView:post called")
+        logger.info(request.body)
+        return JsonResponse({'text': 'get response'})
+
+
+class MockKeystoneUsersProjects(APIView):
+    """
+    This is just a simple view to catch any url we are not handling
+    """
+
+    def get(self, request, user_id):
+        logger.info(" MockKeystoneUserProjects:get called (user id: %s) " % (user_id))
+        # for header in request.META:
+        #    logger.info("   header: %s   value %s" % (header, request.META[header]) )
+        token = request.META.get('HTTP_X_AUTH_TOKEN')
+        logger.info("  token: %s" % token)
+        logger.info("         gAAAAABY2WAKlR6yxvn1mvkPcW0i4fB36y68N0mg2u1CUjPg7OJ-alV2CPPg207_zOTm8s3p3hHJWYVCw5Jh3PBYECkdplVgdzH7UgviSftNM8AJboFNXbypNNaELqgSztpGh5TnBbXI5RlzAc4IIBWd8wPL-RZuOA")
+        logger.info("  user:  %s" % user_id)
+        logger.info("         6705630a653f4300baf486a8df6072de")
+        if token == 'gAAAAABY2WAKlR6yxvn1mvkPcW0i4fB36y68N0mg2u1CUjPg7OJ-alV2CPPg207_zOTm8s3p3hHJWYVCw5Jh3PBYECkdplVgdzH7UgviSftNM8AJboFNXbypNNaELqgSztpGh5TnBbXI5RlzAc4IIBWd8wPL-RZuOA'\
+                and user_id == "6705630a653f4300baf486a8df6072de":
+            json_text = '{"links": {"self": "http://localhost:8082/api/test/keystone/v3/users/6705630a653f4300baf486a8df6072de/projects", "previous": null, "next": null}, "projects":'\
+                        ' [{"is_domain": false, "description": "a test", "links": {"self": "http://localhost:8082/api/test/kestone/v3/projects/190ce9f5a454493e9eaae608d54fe2d1"},'\
+                        ' "enabled": true, "id": "190ce9f5a454493e9eaae608d54fe2d1", "parent_id": "default", "domain_id": "default", "name": "MockeyMock"}]}'
+            response = HttpResponse(json_text, status=status.HTTP_200_OK, content_type="application/json")
+            response['x-openstack-request-id'] = "req-6a536bef-af21-46a9-94e3-31f47998644f"
+        elif token == 'gAAAAABY2WAKlR6yxvn1mvkPcW0i4fB36y68N0mg2u1CUjPg7OJ-alV2CPPg207_zOTm8s3p3hHJWYVCw5Jh3PBYECkdplVgdzH7UgviSftNM8AJboFNXbypNNaELqgSztpGh5TnBbXI5RlzAc4IIBWd8wPL-RZuOA':
```

Comment:
@xuhang57 This is not a uuid. But anyway, this is the token we are returning from mock keystone, therefore it IS the token that atmosphere will use for all communications.